nodejs_package/brainflow/board-shim.ts

```diff
@@ -0,0 +1,232 @@
+    return JSON.stringify(params);
+  }
+}
+
+class BoardControllerDLL extends BoardControllerFunctions {
+  private libPath: string;
+  private dllPath: string;
+  private lib: koffi.IKoffiLib;
+
+  constructor() {
+    super();
+    this.libPath = `${__dirname}/lib`;
+    this.dllPath = this.getDLLPath();
+    this.lib = this.getLib();
+
+    this.isPrepared = this.lib.func(CLike.is_prepared);
+    this.prepareSession = this.lib.func(CLike.prepare_session);
+    this.startStream = this.lib.func(CLike.start_stream);
+    this.getBoardDataCount = this.lib.func(CLike.get_board_data_count);
+    this.getBoardData = this.lib.func(CLike.get_board_data);
+    this.getCurrentBoardData = this.lib.func(CLike.get_current_board_data);
+    this.getNumRows = this.lib.func(CLike.get_num_rows);
+    this.releaseAllSessions = this.lib.func(CLike.release_all_sessions);
+    this.releaseSession = this.lib.func(CLike.release_session);
+    this.stopStream = this.lib.func(CLike.stop_stream);
+    this.getSamplingRate = this.lib.func(CLike.get_sampling_rate);
+    this.getEegChannels = this.lib.func(CLike.get_eeg_channels);
+    this.addStreamer = this.lib.func(CLike.add_streamer);
+  }
+
+  private getDLLPath() {
+    const platform = os.platform();
+    const arch = os.arch();
+    switch (platform) {
+      case 'darwin':
+        return `${this.libPath}/libBoardController.dylib`;
+      case 'win32':
+        return arch === 'x64' ? `${this.libPath}/BoardController.dll` : `${this.libPath}/BoardController32.dll`;
+      case 'linux':
+        return `${this.libPath}/libBoardController.so`;
+      default:
+        throw new Error(`OS ${platform} is not supported.`);
+    }
+  }
+
+  private getLib() {
+    try {
+      const lib = koffi.load(this.dllPath);
+      return lib;
+    } catch (err) {
+      console.error(err);
+      throw new Error(`${'Could not load BoardController DLL - path://'}${this.dllPath}`);
+    }
+  }
+}
+
+export class BoardShim {
+  private boardId: BoardIds;
+
+  private masterBoardId: BoardIds;
+
+  private inputJson: string;
+
+  private boardController: BoardControllerDLL;
+
+  constructor(boardId: BoardIds, inputParams: Partial<IInputParams>) {
+    this.boardId = boardId;
+    this.masterBoardId =
+      inputParams.masterBoard && inputParams.masterBoard !== BoardIds.NO_BOARD ? inputParams.masterBoard : boardId;
+    this.inputJson = new BrainFlowInputParams(inputParams).toJson();
+    this.boardController = new BoardControllerDLL();
+  }
+
+  public isPrepared(): boolean {
+    const prepared = [0];
+    const res = this.boardController.isPrepared(prepared, this.boardId, this.inputJson);
+    if (res !== BrainFlowExitCodes.STATUS_OK) {
+      throw new Error('Could not check prepared state');
+    }
+    return !!prepared[0];
+  }
+
+  public addStreamer(streamerParams: string, preset = BrainFlowPresets.DEFAULT_PRESET): void {
+    const res = this.boardController.addStreamer(streamerParams, preset, this.boardId, this.inputJson);
```

Comment:
hey @Andrey1994 
In python I see we're doing this:

```py
if streamer_params is None:
            streamer = None
        else:
            try:
                streamer = streamer_params.encode()
            except BaseException:
                streamer = streamer_params
```

I don't see anything similar in Java. Not sure to understand why we're doing this.